test/fake-ws.js

```diff
@@ -101,75 +97,44 @@ WebSocket.prototype.abort = function () {
 };
 
 WebSocket.prototype.send = function (data) {
-
     let rec_data = this.decode(data);
-
-    // console.log('Server received a message: ', rec_data);
-    let send_data, enc_data, i, opts, pptSerializer;
-    send_data = lodash.cloneDeep(sendData[sendDataCursor++]);
-
-    if ((rec_data[0] === WAMP_MSG_SPEC.CALL ||
-            rec_data[0] === WAMP_MSG_SPEC.PUBLISH ||
-            rec_data[0] === WAMP_MSG_SPEC.YIELD) &&
-        send_data.data &&
-        (send_data.data[0] === WAMP_MSG_SPEC.EVENT ||
-            send_data.data[0] === WAMP_MSG_SPEC.RESULT ||
-            send_data.data[0] === WAMP_MSG_SPEC.INVOCATION)) {
-
-        if (send_data.data[0] === WAMP_MSG_SPEC.EVENT) {
-            opts = send_data.data[3];
-        } else if (send_data.data[0] === WAMP_MSG_SPEC.RESULT) {
-            opts = send_data.data[2];
-        } else if (send_data.data[0] === WAMP_MSG_SPEC.INVOCATION) {
-            opts = send_data.data[3];
+    let enc_data, i, options, pptSerializer;
+    let send_data = lodash.cloneDeep(sendData[sendDataCursor++]);
+
+    if (
+        ([WAMP_MSG_SPEC.CALL, WAMP_MSG_SPEC.PUBLISH, WAMP_MSG_SPEC.YIELD].includes(rec_data?.[0])) &&
+        ([WAMP_MSG_SPEC.EVENT, WAMP_MSG_SPEC.RESULT, WAMP_MSG_SPEC.INVOCATION].includes(send_data?.data?.[0]))
+    ) {
+        if ([WAMP_MSG_SPEC.EVENT, WAMP_MSG_SPEC.INVOCATION].includes(send_data.data[0])) {
+            options = send_data.data[3];
+        } else if ([WAMP_MSG_SPEC.RESULT].includes(send_data.data[0])) {
+            options = send_data.data[2];
         }
 
-        // Check for PPT mode and encode payload with appreciate serializer
-        if (opts.ppt_scheme &&
-            opts.ppt_serializer &&
-            opts.ppt_serializer !== 'native') {
-            switch (opts.ppt_serializer) {
-                case 'cbor':
-                    pptSerializer = new CborSerializer();
-                    break;
-                case 'msgpack':
-                    pptSerializer = new MsgpackSerializer();
-                    break;
-                case 'json':
-                    pptSerializer = new JsonSerializer();
-                    break;
-            }
+        // Check for PPT mode and encode payload with serializer
+        if (options.ppt_scheme &&
+            options.ppt_serializer &&
+            options.ppt_serializer !== 'native') {
+            pptSerializer = serializers[options.ppt_serializer];
+
+            if ([WAMP_MSG_SPEC.EVENT, WAMP_MSG_SPEC.INVOCATION].includes(send_data.data[0])) {
+                const payload = pptSerializer.encode(send_data.data[4][0]);
+                const ruinedPayload = String(payload);
```

Comment:
Is this really break any payload independent of the serializer used?

Reply:
I forgot the '123' at the end, I've added them back